### PR TITLE
Base class implementation.

### PR DESCRIPTION
**Pull request by misohu**
This PR implements this [upstream component](https://github.com/kubeflow/kubeflow/blob/master/components/pvcviewer-controller/controllers/pvcviewer_controller.go#L55). Key features:
- Implemented in base charm.
- On every init it generates ssl files if they are not presented in the _store. These certificates are then pushed to container and the ca file is provided to `webhook` objects. We use similar implementation as in `kserve-operators`

Integration tests:
- we deploy istio and istio-pilot plus the pvcviewer-charm
- we try the upstream example (to match upstream testing)

### CODE REVIEW

File: src/components/pebble_component.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class PvcViewerPebbleService(PebbleServiceComponent):
+    def get_layer(self) -> Layer:
+        """Defines and returns Pebble layer configuration
+
+        This method is required for subclassing PebbleServiceContainer
+        """
+        logger.info("PebbleServiceComponent.get_layer executing")
+        return Layer(
+            {
+                "summary": "kfp-viewer layer",
```
Comment (i-chvets) [inline src/components/pebble_component.py:20]: Should it be `pvc-viewer`?

File: src/components/pebble_component.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                "description": "Pebble config layer for pvcviewer",
+                "services": {
+                    self.service_name: {
+                        "override": "replace",
+                        "summary": "Entry point for pvcviewer image",
+                        "command": "/manager --leader-elect",
+                        "startup": "enabled",
+                        "environment": {},
```
Comment (i-chvets) [inline src/components/pebble_component.py:28]: May be we should omit `environment` if it is empty.

**Comment by i-chvets** [comment]
Should we also add removal test in which we test accessibility of URL after charm is removed?
Just to make sure we clean up everything.

File: .github/workflows/integrate.yaml [hunk 3]
```diff
@@ -73,7 +73,7 @@ jobs:
         microk8s-addons: "dns hostpath-storage rbac metallb:10.64.140.43-10.64.140.49"
 
     - name: Run integration tests
-      run: tox -vve integration -- --model kubeflow
+      run: sg snap_microk8s -c "tox -vve integration -- --model kubeflow"
```
Comment (i-chvets) [inline .github/workflows/integrate.yaml:76]: Do we want to set up specific versions of microk8s and juju for this?

**Comment by misohu** [comment]
Hi @i-chvets resolved your comments ... added integration test for remove application. I also added readme.

**Comment by i-chvets** [comment]
Failed integration test in local setup. Timeout on waiting for UI.
```
INFO     httpx:_client.py:1013 HTTP Request: PATCH https://10.1.1.124:16443/api/v1/namespaces/kubeflow-user-example-com?fieldManager=pvcviewer-operator "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1013 HTTP Request: PATCH https://10.1.1.124:16443/apis/kubeflow.org/v1alpha1/namespaces/kubeflow-user-example-com/pvcviewers/pvcviewer-sample?fieldManager=pvcviewer-operator "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1013 HTTP Request: PATCH https://10.1.1.124:16443/api/v1/namespaces/kubeflow-user-example-com/persistentvolumeclaims/pvcviewer-sample?fieldManager=pvcviewer-operator "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1013 HTTP Request: GET https://10.1.1.124:16443/api/v1/namespaces/test-charm-fwxe/services/istio-ingressgateway-workload "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1013 HTTP Request: PATCH https://10.1.1.124:16443/api/v1/namespaces/kubeflow-user-example-com?fieldManager=pvcviewer-operator "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1013 HTTP Request: PATCH https://10.1.1.124:16443/apis/kubeflow.org/v1alpha1/namespaces/kubeflow-user-example-com/pvcviewers/pvcviewer-sample?fieldManager=pvcviewer-operator "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1013 HTTP Request: PATCH https://10.1.1.124:16443/api/v1/namespaces/kubeflow-user-example-com/persistentvolumeclaims/pvcviewer-sample?fieldManager=pvcviewer-operator "HTTP/1.1 200 OK"
INFO     httpx:_client.py:1013 HTTP Request: GET https://10.1.1.124:16443/api/v1/namespaces/test-charm-fwxe/services/istio-ingressgateway-workload "HTTP/1.1 200 OK"
FAILED
```

**Comment by i-chvets** [comment]
Verified by creating PR https://github.com/canonical/pvcviewer-operator/pull/7

**Review by i-chvets — APPROVED** [review]
Approved based on verification in GH runners.

**Pull request merged**